db/sovereign-db/src/state_db.rs

```diff
@@ -27,8 +28,14 @@ impl StateDB {
             STATE_TABLES.iter().copied(),
             &gen_rocksdb_options(&Default::default(), false),
         )?;
+
+        let version = Self::last_version(&inner)?
+            .map(|v| v + 1)
+            .unwrap_or_default();
+
```

Comment:
Thinking about this more, I'm afraid that the semantics might be confusing. Is `version` the next version to write, or the last version written? 

Personally, I think `last_version_written` is probably preferable, since there might be circumstances where people write to the same version multiple times (for example, when recovering from a crash).

Reply:
I have changed it to `last_version_written`